terraform/lib/dependabot/terraform/file_parser.rb

```diff
@@ -68,7 +68,11 @@ def parse_terragrunt_files(dependency_set)
           modules.each do |details|
             next unless details["source"]
 
-            dependency_set << build_terragrunt_dependency(file, details)
+            source = source_from(details)
+            # Cannot update nil (interpolation sources) or local path modules, skip
+            next if source.nil? || source[:type] == "path"
```

Comment:
I don't know how this code is used, and don't know Terragrunt/Terraform very well.

Obviously we can't update local path modules as deps to the current terragrunt files, but is there any risk that this prevents updating a local path module that's sitting in the same monorepo? Ie, because this stops :dependabot: from parsing the local module, it also prevent Dependabot from updating any remote deps listed in _that_ local module?

I highly doubt that's a risk, so I'm going to merge, but wanted to doublecheck with you @dwc0011 since you're the expert.

Reply:
I don't think it would _prevent_ it, exactly... There simply is no way for Dependabot to know how the Terragrunt interpolation resolves. The resolution depends on the Terragrunt config at runtime, and may not even be _consistent_. So when the user is using Terragrunt interpolation this way, Dependabot simply _can't_ use its recursive feature to update anything in that other path anyway.

The workaround for the user would be just to specify both paths in the Dependabot config, so then Dependabot processes both. Which honestly is my preference anyway; the recursive feature in Dependabot has been a bit of a pain and I'd rather just disable it. (Though now with grouping it is becoming of more interest...)

Reply:
For terraform local path modules, dependabot does not support updating local path modules, so it makes sense that terragrunt local path modules would not be updated as well. 
Local Path Module Files are found and traversed by the file fetcher, not the file parser, all files including those where local path modules are found are then passed to the file parser. That said Terragrunt files do not have a fetch local path module at this time. (This could be added in a separate PR)

As mentioned, the file fetcher is responsible for gathering all the files to process including local source module files.  
For terraform files see: [fetch local path modules](https://github.com/dependabot/dependabot-core/blob/2530fd02738ddfd6b6251a736953c18946f7ec71/terraform/lib/dependabot/terraform/file_fetcher.rb#L54C11-L54C34 ). 


Since the process is fetch the files, parse, check updatable, and then update; not returning terragrunt local path dependency in the file_parser makes the most sense.  We do this for other types that are not supported as well. i.e. local path modules for terraform, interpolation for terragrunt (added by this PR), 


Even if the dependency was added in the file parser, it would be removed later in the [update checker](https://github.com/dependabot/dependabot-core/blob/2530fd02738ddfd6b6251a736953c18946f7ec71/terraform/lib/dependabot/terraform/update_checker.rb#L193)

Reply:
Thanks both of you. That's very helpful context, and I appreciate the extra details... Will be useful for any future `git blame` spelunkers.